src/expr.rs

```diff
@@ -298,9 +308,57 @@ macro_rules! numeric (
 	($i:expr, $f:expr ) => (map_opt!($i,call!($f),EvalResult::as_numeric));
 );
 
+enum TypeCast {
+	SignedInt(usize),
+	UnsignedInt(usize),
+}
+
+impl EvalResult {
+	fn cast(self, ctype: TypeCast) -> Self {
+		use self::TypeCast::*;
+		let round = |bits| 2i64.pow((bits) as u32);
```

Comment:
I think this doesn't implement rounding, maybe give it another name?

Reply:
Is `pow2` a proper name?